service/history/workflow/history_builder.go

```diff
@@ -729,6 +729,21 @@ func (b *HistoryBuilder) AddUpsertWorkflowSearchAttributesEvent(
 	return b.appendEvents(event)
 }
 
+func (b *HistoryBuilder) AddUpsertWorkflowMemoEvent(
+	workflowTaskCompletedEventID int64,
+	command *commandpb.UpsertWorkflowMemoCommandAttributes,
+) *historypb.HistoryEvent {
+	event := b.createNewHistoryEvent(enumspb.EVENT_TYPE_UPSERT_WORKFLOW_MEMO, b.timeSource.Now())
+	event.Attributes = &historypb.HistoryEvent_UpsertWorkflowMemoEventAttributes{
```

Comment:
I thought we will use event defined in https://github.com/temporalio/api/pull/196?

Reply:
This is why it is draft. It will use those eventually.